feat(scheduler-targets): step function start execution target

WtfJoke opened the pull request:
A StepFunctionStartExecution ScheduleTarget was implemented similar to the already existing [LambdaInvoke](https://github.com/aws/aws-cdk/blob/b2a895ef285e5451e64c21e179172e998c479582/packages/%40aws-cdk/aws-scheduler-targets-alpha/lib/lambda-invoke.ts#L8) target.

I've added an integration test, which will trigger an eventbridge schedule once. This schedule has a step function as a target. The step function creates a parameter with a given value, which will get verified by the test

Closes #27377 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 7de25648e5bae6f1f8c5fdf80ca765b33088f747
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6c18a773-23b6-4c60-af7b-3b1bd29badd8%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 955607489847b631716546ea68e5f807e3c26288
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=00cd560f-7383-4a0e-8e56-9f7d258ff2b6%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 01ff0cb61d6318c51057680f772d4f584823d507
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=62460766-f4b7-48f7-8793-e321f19dda6a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: b8f00dd77317cc441f833b9c45a1fce841823994
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=eb8d3e97-6896-4f76-9a18-e9f1c1632c18%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

filletofish commented on packages/cdk-cli-wrapper/.warnings.jsii.js:
```diff
@@ -0,0 +1,73 @@
+function cdk_cli_wrapper_ICdk(p) {
```
Looks like cdk-cli-wrapper folder was committed by mistake?

WtfJoke replied:
Yeah, I commited this by mistake. Good spot! Thanks

filletofish commented on packages/@aws-cdk/aws-scheduler-targets-alpha/lib/stepfunctions-start-execution.ts:
```diff
@@ -0,0 +1,35 @@
+/**
+ * Use an AWS Step function as a target for AWS EventBridge Scheduler.
+ */
+export class StepFunctionsStartExecution extends ScheduleTargetBase implements IScheduleTarget {
```
I would recommend to add unit tests for this target too. See https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-scheduler-targets-alpha/test/target.test.ts#L9. We probably need to rename that test file and create one file with unit tests per each target.

WtfJoke replied:
I created a new unit test which contains almost identical unit tests for step functions.

I also renamed the existing file `target.test.ts` into `lambda-invoke.test.ts`.

filletofish reviewed (changes requested):
Thank you for the commit! It motivated me to add more issues for Scheduler Targets to get help from the community to build all targets for L2 Scheduler constructs.

WtfJoke commented:
@filletofish thanks for the review! I'm glad you created more issues for it, I thought about contributing more targets, but have chosen the one I needed the most :)

I've addressed your comments, please have a look on it again. 
Thank you!

WtfJoke commented on packages/@aws-cdk/aws-scheduler-targets-alpha/lib/stepfunctions-start-execution.ts:
```diff
@@ -0,0 +1,35 @@
+  }
+
+  protected addTargetActionToRole(schedule: ISchedule, role: IRole): void {
+    const stateMachineEnv = this.stateMachine.env;
+    if (!sameEnvDimension(stateMachineEnv.region, schedule.env.region)) {
+      throw new Error(`Cannot assign stateMachine in region ${stateMachineEnv.region} to the schedule ${Names.nodeUniqueId(schedule.node)} in region ${schedule.env.region}. Both the schedule and the stateMachine must be in the same region.`);
+    }
+
+    if (!sameEnvDimension(stateMachineEnv.account, schedule.env.account)) {
+      throw new Error(`Cannot assign stateMachine in account ${stateMachineEnv.account} to the schedule ${Names.nodeUniqueId(schedule.node)} in account ${schedule.env.region}. Both the schedule and the stateMachine must be in the same account.`);
+    }
+
+    if (this.props.role && !sameEnvDimension(this.props.role.env.account, stateMachineEnv.account)) {
+      throw new Error(`Cannot grant permission to execution role in account ${this.props.role.env.account} to invoke target ${Names.nodeUniqueId(this.stateMachine.node)} in account ${stateMachineEnv.account}. Both the target and the execution role must be in the same account.`);
+    }
```
It might make sense to extract this common logic block somewhere, do you have a favorite spot for it? Should it go to `ScheduleTargetBase` or in a new class/function?

filletofish replied:
@kaizencc do you have any opinion on this? I personally do not mind the duplication of the code

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 0c4b89884e85499b981afdad69941df9b2dec1cf
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=9e20c39e-6144-4047-9ded-a7df3e8bf28e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: ae2e8ae85f10f0cd472eaaa2f5bd88bee5bd6758
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=09250ce3-960b-45ab-ac6d-e1b9cbd724a4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

kaizencc reviewed (approved):
Thanks for the submission @WtfJoke!

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: d3302f00f8b83f5bd107d696e4bf8aa483365050
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=90eea760-4d9c-4e02-a4eb-74c0222d112b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->
